File: javascript/calculator.js
'use strict'

const readline = require('readline')

let firstNumber = 0
let secondNumber = 0

const divide = (firstNumber, secondNumber) => {
  return firstNumber /secondNumber 
}

const multiply = (firstNumber, secondNumber) => {
  return firstNumber *secondNumber 
}

const subtract = (firstNumber, secondNumber) => {
  return firstNumber -secondNumber 
}

const add = (firstNumber, secondNumber) => {
  return firstNumber +secondNumber 
}

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
})

const getFirstNumber = () => {
  return new Promise((resolve, reject) => {
    rl.question('Enter your first number: ', number => {
      firstNumber = parseInt(number)
      resolve()
    })
  })
}

const getSecondNumber = () => {
  return new Promise((resolve, reject) => {
    rl.question('Enter your second number: ', number => {
      secondNumber= parseInt(number)
      resolve()
    })
  })
}

const makeOperation = () => {
  return new Promise((resolve, reject) => {
    console.log('Current supported operations: ')
    console.log('(1) +')
    console.log('(2) -')
    console.log('(3) *')
    console.log('(4) /')
    rl.question('Select the operation number: ', operation => {
      switch (operation) {
        case '1':
          console.log(add(firstNumber, secondNumber))
          break
        case '2':
          console.log(subtract(firstNumber, secondNumber))
          break
        case '3':
          console.log(multiply(firstNumber, secondNumber))
          break
        case '4':
          console.log(divide(firstNumber, secondNumber))
          break
        default:
          console.log('chose a valid operation')
          break
      }
      resolve()
    })
  })
}

const main = async () => {
  await getFirstNumber()
  await getSecondNumber()
  await makeOperation()
  rl.close()
}

main()

File: Python/calculator.py
def divide(number1, number2):
  return number1 / number2

def multiply(number1, number2):
  return number1 * number2

def subtract(number1, number2):
  return number1 - number2

def sum(number1, number2):
  return number1 + number2

def main():
  print("Calculator")
  number1 = int(input("Enter your first number: "))
  number2 = int(input("Enter your second number: "))

  print("Current supported operations:")
  print("(1) +")
  print("(2) -")
  print("(3) *")
  print("(4) /")
  operation = int(input("Enter the number of the operation: "))

  if operation == 1:
    print(sum(number1, number2))
  elif operation == 2:
    print(subtract(number1, number2))
  elif operation == 3:
    print(multiply(number1, number2))
  elif operation == 4:
    print(divide(number1, number2))

if __name__ == "__main__":
  main()